File: gto-support-core/src/main/java/org/ccci/gto/android/common/support/v4/util/FragmentUtils.java
package org.ccci.gto.android.common.support.v4.util;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

/**
 * @deprecated Since v4.1.0, use {@link org.ccci.gto.android.common.androidx.fragment.app.FragmentKt} instead.
 */
@Deprecated
public final class FragmentUtils {
    private FragmentUtils() {}

    /**
     * @deprecated
     *    Since v4.1.0, use {@link org.ccci.gto.android.common.androidx.fragment.app.FragmentKt#findAncestorFragment()}
     *    instead.
     */
    @Deprecated
    @Nullable
    public static <T> T getAncestorFragment(@NonNull final Fragment fragment, @NonNull final Class<T> clazz) {
        Fragment parent = fragment.getParentFragment();
        while (parent != null) {
            if (clazz.isInstance(parent)) {
                return clazz.cast(parent);
            }
            parent = parent.getParentFragment();
        }
        return null;
    }

    /**
     * @deprecated
     *    Since v4.1.0, use {@link org.ccci.gto.android.common.androidx.fragment.app.FragmentKt#findListener()} instead.
     */
    @Deprecated
    @Nullable
    public static <T> T getListener(@NonNull final Fragment fragment, @NonNull final Class<T> clazz) {
        final Fragment frag = fragment.getParentFragment();
        if (clazz.isInstance(frag)) {
            return clazz.cast(frag);
        }

        final FragmentActivity activity = fragment.getActivity();
        if (clazz.isInstance(activity)) {
            return clazz.cast(activity);
        }

        return null;
    }
}

File: gto-support-core/src/main/java/org/ccci/gto/android/common/util/FragmentUtils.kt
package org.ccci.gto.android.common.util

import androidx.fragment.app.Fragment
import org.ccci.gto.android.common.androidx.fragment.app.findAncestorFragment
import org.ccci.gto.android.common.androidx.fragment.app.findListener

@Deprecated(
    "Since v3.6.2, use findAncestorFragment() from gto-support-androidx-fragment module instead",
    ReplaceWith("findAncestorFragment<T>()", "org.ccci.gto.android.common.androidx.fragment.app.findAncestorFragment")
)
inline fun <reified T> Fragment.findAncestorFragment() = findAncestorFragment<T>()

@Deprecated(
    "Since v3.6.2, use findListener() from gto-support-androidx-fragment module instead",
    ReplaceWith("findListener()", "org.ccci.gto.android.common.androidx.fragment.app.findListener")
)
inline fun <reified T> Fragment.findListener() = findListener<T>()
